tools/update_assisted_installer_yaml.py

```diff
@@ -11,22 +12,56 @@
 args = parser.parse_args()
 
 
+def in_all_lists(el: str, lists: list[str]) -> bool:
+    for list in lists:
+        if el not in list:
+            return False
+    return True
+
+
+def find_first_common_element(lists: list[str]) -> str:
+    if len(lists) < 1:
+        raise IndexError("No lists provided. Cannot find common element")
+    for el in lists[0]:
+        if in_all_lists(el, lists[1:]):
+            return el
+    raise RuntimeError("Could not find common element")
+
+
+def get_ref_by_docker_image(images):
+    tag_pattern = "^latest-[a-f0-9]{40}$"
+    tag_prefix = "latest-"
+    sk = skopeo.Skopeo()
+    image_tags = []
+    for img in images:
+        image_tags.append(sk.get_image_tags_by_pattern(img, tag_pattern))
+    if len(image_tags) < 1:
+        raise IndexError(f"Could not find any tags matching pattern `{tag_pattern}` for images `{images}`")
+    if len(image_tags) == 1:
+        tag = image_tags[0][0]
+    else:
+        tag = find_first_common_element(image_tags)
+    return tag.removeprefix(tag_prefix)
+
+
 def main():
     full_release = args.full or os.environ.get("NIGHTLY_RELEASE", "false") == "true"
 
     with open(args.deployment, "r") as f:
         deployment = yaml.safe_load(f)
 
     for rep in deployment:
-        if full_release:
-            hash = subprocess.check_output(f"git ls-remote https://github.com/{rep}.git HEAD | cut -f 1", shell=True)[:-1]
-            hash = hash.decode("utf-8")
-        else:
-            hash = os.environ.get(rep, None)
+        try:
+            if full_release:
+                hash = get_ref_by_docker_image(deployment[rep]['images'])
+            else:
+                hash = os.environ.get(rep, None)
             if not hash:
                 continue
-
-        update_hash.update_hash(deployment_yaml=args.deployment, repo=rep, hash=hash)
+            update_hash.update_hash(deployment_yaml=args.deployment, repo=rep, hash=hash)
+        except LookupError as e:
+            print(e, file=sys.stderr)
+            sys.exit(1)
```

Comment:
seems to be equivalent to just not ``try-excepting`` and letting it fail for this exception